composition/src/errors/errors.ts

```diff
@@ -415,6 +451,19 @@ export function unexpectedParentKindErrorMessage(
   return ` Expected "${parentTypeName}" to be type ${expectedTypeString} but received "${actualTypeString}".`;
 }
 
+export function unexpectedParentKindForChildError(
+  parentTypeName: string,
+  expectedTypeString: string,
+  actualTypeString: string,
+  childName: string,
+  childTypeString: string,
+): Error {
+  return new Error(
```

Comment:
For a future version, we should throw concrete error types to distinguish between composition and other errors.

Reply:
I have a decent plan for a whole error overhaul. Will discuss with you at some point.